### DISCUSSION TITLE
Change of RAS implementation

### DISCUSSION THREAD

**Discussion by alexzubic** [polatory/polatory#185] [General]
Hi, my name is Alex Zubic, a Software Engineer from Sydney Australia. I had a question about the main branch for this repository.

I was wondering why the RAS preconditioner (ras_preconditioner.hpp) implementation was changed from the multi-level to the multigrid type method (compared to the v0 branch). Was there a specific reason for this?

thanks

**Comment by unageek** [comment]
Hi!

IIRC, it was because filtering out low-frequency components first improved convergence rate, and it did not affect the overall firing time. Certain data sets could not be fit without this modification.
